local/network.go

```diff
@@ -737,3 +592,219 @@ func createFileAndWrite(path string, contents []byte) error {
+			log.Info(
+				"not overwriting node config flag %s (value %v) with network config flag (value %v)",
+				flagName, val, flagVal,
+			)
+		}
+	}
+}
+
+// Set [nodeConfig].Name if it isn't given and assert it's unique.
+func (ln *localNetwork) setName(nodeConfig *node.Config) error {
+	// If no name was given, use default name pattern
+	if len(nodeConfig.Name) == 0 {
+		nodeConfig.Name = fmt.Sprintf("%s%d", defaultNodeNamePrefix, ln.nextNodeSuffix)
+		ln.nextNodeSuffix++
+	}
+	// Enforce name uniqueness
+	if _, ok := ln.nodes[nodeConfig.Name]; ok {
+		return fmt.Errorf("repeated node name %q", nodeConfig.Name)
+	}
+	return nil
+}
+
+func makeNodeDir(log logging.Logger, rootDir, nodeName string) (string, error) {
+	if rootDir == "" {
+		log.Warn("no network root directory defined; will create this node's runtime directory in working directory")
+	}
+	// [nodeRootDir] is where this node's config file, C-Chain config file,
+	// staking key, staking certificate and genesis file will be written.
+	// (Other file locations are given in the node's config file.)
+	// TODO should we do this for other directories? Profiles?
+	nodeRootDir := filepath.Join(rootDir, nodeName)
+	if err := os.Mkdir(nodeRootDir, 0o755); err != nil {
+		return "", fmt.Errorf("error creating temp dir: %w", err)
+	}
+	return nodeRootDir, nil
+}
+
+// getConfigEntry returns an entry in the config file if it is found, otherwise returns the default value
+func getConfigEntry(configFile map[string]interface{}, flag string, defaultVal string) (string, error) {
+	var entry string
+	if val, ok := configFile[flag]; ok {
+		if entry, ok := val.(string); ok {
+			return entry, nil
+		}
+		return "", fmt.Errorf("expected flag %q to be string but got %T", flag, entry)
+	}
+	return defaultVal, nil
+}
+
+// getPort looks up the port config in the config file, if there is none, it tries to get a random free port from the OS
+func getPort(configFile map[string]interface{}, flag string) (port uint16, err error) {
+	var entry uint16
+	if val, ok := configFile[flag]; ok {
+		if entry, ok := val.(float64); ok {
+			return uint16(entry), nil
+		}
+		return 0, fmt.Errorf("expected flag %q to be string but got %T", flag, entry)
+	}
+	// Use a random free port.
+	// Note: it is possible but unlikely for getFreePort to return the same port multiple times.
+	port, err = getFreePort()
+	if err != nil {
+		return 0, fmt.Errorf("couldn't get free API port: %w", err)
+	}
+	return port, nil
+}
+
+// nodeInfo is only internally used so we can pass some node information between funcs
+type nodeInfo struct {
+	flags   []string
+	apiPort uint16
+	p2pPort uint16
+	nodeID  ids.ShortID
+}
+
+// buildFlags returns the:
+// 1) Flags
+// 2) API port
+// 3) P2P port
+// 4) Node ID
+// of the node being added with config [nodeConfig], config file [configFile],
+// and directory at [nodeDir].
+func (ln *localNetwork) buildFlags(configFile map[string]interface{}, nodeDir string, nodeConfig *node.Config) (nodeInfo, error) {
```

Comment:
should have preferred this function to receive ports (or to receive partial localNode) and return flags, avoiding nodeInfo struct (other function to generate the node ports). name: buildNodeFlags